builtin/providers/aws/resource_aws_lambda_event_source_mapping.go

```diff
@@ -0,0 +1,150 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/lambda"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsLambdaEventSourceMapping() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsLambdaEventSourceMappingCreate,
+		Read:   resourceAwsLambdaEventSourceMappingRead,
+		Update: resourceAwsLambdaEventSourceMappingUpdate,
+		Delete: resourceAwsLambdaEventSourceMappingDelete,
+
+		Schema: map[string]*schema.Schema{
+			"function_arn": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"event_source_arn": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"starting_position": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"batch_size": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  100,
+			},
+			"uuid": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"last_modified": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"last_processing_result": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"state": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"state_transition_reason": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+// resourceAwsLambdaEventSourceMappingCreate maps to:
+// CreateEventSourceMapping in the API / SDK
+func resourceAwsLambdaEventSourceMappingCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).lambdaconn
+
+	functionName := d.Get("function_arn").(string)
+	eventSourceArn := d.Get("event_source_arn").(string)
+	startingPosition := d.Get("starting_position").(string)
+	batchSize := d.Get("batch_size").(int)
+
+	log.Printf("[DEBUG] Creating Lambda event source mapping: source %s to function %s", eventSourceArn, functionName)
+
+	// NOTE: Enabled is omitted here but defaults to true
+	params := &lambda.CreateEventSourceMappingInput{
+		FunctionName: aws.String(functionName),
```

Comment:
There are some [go fmt](https://blog.golang.org/go-fmt-your-code) issues in this block.